cmd/experimental/kjobctl/pkg/cmd/delete/delete_interactive.go

```diff
@@ -0,0 +1,175 @@
+	"sigs.k8s.io/kueue/cmd/experimental/kjobctl/pkg/cmd/util"
+	"sigs.k8s.io/kueue/cmd/experimental/kjobctl/pkg/constants"
+)
+
+const (
+	interactiveExample = `  # Delete interactive shell 
+  kjobctl delete interactive my-application-profile-interactive-k2wzd`
+)
+
+type InteractiveOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+
+	InteractiveNames []string
+	Namespace        string
+
+	CascadeStrategy metav1.DeletionPropagation
+	DryRunStrategy  util.DryRunStrategy
+
+	Client corev1.CoreV1Interface
+
+	PrintObj printers.ResourcePrinterFunc
+
+	genericiooptions.IOStreams
+}
+
+func NewInteractiveOptions(streams genericiooptions.IOStreams) *InteractiveOptions {
+	return &InteractiveOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("deleted").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewInteractiveCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewInteractiveOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:                   "interactive NAME [--cascade STRATEGY] [--dry-run STRATEGY]",
+		DisableFlagsInUseLine: true,
+		Short:                 "Delete interactive shell",
+		Example:               interactiveExample,
+		Args:                  cobra.MinimumNArgs(1),
+		ValidArgsFunction:     completion.PodNameFunc(clientGetter),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+
+			err := o.Complete(clientGetter, cmd, args)
+			if err != nil {
+				return err
+			}
+
+			return o.Run(cmd.Context())
+		},
+	}
+
+	addCascadingFlag(cmd)
+	util.AddDryRunFlag(cmd)
+
+	o.PrintFlags.AddFlags(cmd)
+
+	return cmd
+}
+
+func (o *InteractiveOptions) Complete(clientGetter util.ClientGetter, cmd *cobra.Command, args []string) error {
+	o.InteractiveNames = args
+
+	var err error
+
+	o.Namespace, _, err = clientGetter.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.DryRunStrategy, err = util.GetDryRunStrategy(cmd)
+	if err != nil {
+		return err
+	}
+
+	err = util.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
+	if err != nil {
+		return err
+	}
+
+	o.CascadeStrategy, err = getCascadingStrategy(cmd)
+	if err != nil {
+		return err
+	}
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	o.PrintObj = printer.PrintObj
+
+	clientset, err := clientGetter.K8sClientset()
+	if err != nil {
+		return err
+	}
+
+	o.Client = clientset.CoreV1()
+
+	return nil
+}
+
+func (o *InteractiveOptions) Run(ctx context.Context) error {
+	for _, podName := range o.InteractiveNames {
+		pod, err := o.Client.Pods(o.Namespace).Get(ctx, podName, metav1.GetOptions{})
+		if client.IgnoreNotFound(err) != nil {
+			return err
+		}
+		if err != nil {
+			fmt.Fprintln(o.ErrOut, err)
+			continue
+		}
+		if _, ok := pod.Labels[constants.ProfileLabel]; !ok {
+			fmt.Fprintln(o.ErrOut, errors.NewNotFound(schema.GroupResource{Group: "v1", Resource: "pod"}, podName).Error())
```

Comment:
Maybe this can be confusing to users.... It's not that the pod doesn't exist, it is that it's not a Pod created via kjob.
Could we change the error message in that case?

Reply:
Done